rpc-rs/src/otel.rs

```diff
@@ -93,13 +105,10 @@ impl Injector for OtelHeaderInjector {
         // probably ok for now as it is freed at the end, but I prefer telemetry stuff to be as
         // little overhead as possible. If anyone has a better idea of how to handle this, please PR
         // it in
-        let header_name = key.escape_default().to_string().into_bytes();
-        let escaped = value.escape_default().to_string().into_bytes();
         // SAFETY: All chars escaped above
-        self.inner.insert(
-            async_nats::header::HeaderName::from_bytes(&header_name).unwrap(),
-            async_nats::HeaderValue::from_bytes(&escaped).unwrap(),
-        );
+        let key = key.escape_default().to_string();
```

Comment:
The new `HeaderMap` doesn't require any escaping anymore (as it just converts from borrowed strings). So this can be removed and `key` just passed straight in